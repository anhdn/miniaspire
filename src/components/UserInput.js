import React  from 'react'
import useForm from './useForm'

import styled from 'styled-components'

const BankContainer = styled.div`
padding: 10px 10px;
margin-bottom: 20px;
`
const Title = styled.h2`
color: #292a2a;
font-size: 14px;
font-weight: 600;
margin-bottom: 9px;
`
const UserInput = ({ onSelectedUserInfo }) => {
  // eslint-disable-next-line
  const { values, handleChange, handleSubmit } = useForm(onSelectedUserInfo)

  return (
    <BankContainer>
      <Title>Tell us about yourself</Title>
      <div className='form-group'>
        <label htmlFor='businessName'>Full Legal Name :</label>
        <input
          type='text'
          className='form-control'
          id='businessName'
          placeholder='Full name as per your ID card'
          onChange={handleChange}
          value={values.businessName || ''}
          name='businessName'
          required
        />
      </div>
      <div className='form-group'>
        <label htmlFor='phoneNumber'>Mobile Phone*</label>
        <div className='input-group mb-2 mr-sm-2 mb-sm-0'>
          <div className='input-group-prepend'>
            <span className='input-group-text' id='basic-addon1'>+65</span>
          </div>
          <input
            type='text'
            className='form-control'
            id='phoneNumber'
            name='phoneNumber'
            placeholder='91234567'
            onChange={handleChange}
            value={values.phoneNumber || ''}
            required
          />
        </div>
      </div>
      <div className='form-group'>
        <label htmlFor='businessNumber'>Business Registration Number (UEN)*:</label>
        <input
          type='text'
          className='form-control'
          id='businessNumber'
          name='businessNumber'
          placeholder='Fill Business Registration Number'
          onChange={handleChange}
          value={values.businessNumber || ''}
          required
        />
      </div>
    </BankContainer>
  )
}

export default UserInput
