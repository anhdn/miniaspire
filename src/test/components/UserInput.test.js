import React from 'react'
import { configure } from 'enzyme'
import UserInput from '../../components/UserInput'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

it('renders correctly without function handle change', () => {
  const tree = renderer
    .create(<UserInput />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
