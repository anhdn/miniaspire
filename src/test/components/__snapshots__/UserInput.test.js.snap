// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with function handle change 1`] = `
<div
  className="UserInput__BankContainer-sc-1r0dns6-0 iReAWk"
>
  <h2
    className="UserInput__Title-sc-1r0dns6-1 cAYqQh"
  >
    Tell us about yourself
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="businessName"
    >
      Full Legal Name :
    </label>
    <input
      className="form-control"
      id="businessName"
      name="businessName"
      onChange={[Function]}
      placeholder="Full name as per your ID card"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="phoneNumber"
    >
      Mobile Phone*
    </label>
    <div
      className="input-group mb-2 mr-sm-2 mb-sm-0"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
          id="basic-addon1"
        >
          +65
        </span>
      </div>
      <input
        className="form-control"
        id="phoneNumber"
        name="phoneNumber"
        onChange={[Function]}
        placeholder="91234567"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="businessNumber"
    >
      Business Registration Number (UEN)*:
    </label>
    <input
      className="form-control"
      id="businessNumber"
      name="businessNumber"
      onChange={[Function]}
      placeholder="Fill Business Registration Number"
      required={true}
      type="text"
    />
  </div>
</div>
`;

exports[`renders correctly without function handle change 1`] = `
<div
  className="UserInput__BankContainer-sc-1r0dns6-0 iReAWk"
>
  <h2
    className="UserInput__Title-sc-1r0dns6-1 cAYqQh"
  >
    Tell us about yourself
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="businessName"
    >
      Full Legal Name :
    </label>
    <input
      className="form-control"
      id="businessName"
      name="businessName"
      onChange={[Function]}
      placeholder="Full name as per your ID card"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="phoneNumber"
    >
      Mobile Phone*
    </label>
    <div
      className="input-group mb-2 mr-sm-2 mb-sm-0"
    >
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
          id="basic-addon1"
        >
          +65
        </span>
      </div>
      <input
        className="form-control"
        id="phoneNumber"
        name="phoneNumber"
        onChange={[Function]}
        placeholder="91234567"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="businessNumber"
    >
      Business Registration Number (UEN)*:
    </label>
    <input
      className="form-control"
      id="businessNumber"
      name="businessNumber"
      onChange={[Function]}
      placeholder="Fill Business Registration Number"
      required={true}
      type="text"
    />
  </div>
</div>
`;
